src/rpc_mock.rs

```diff
@@ -0,0 +1,74 @@
+// Implementation of RpcRequestHandler trait for testing Rpc requests without i/o
+
+use crate::rpc_request::{RpcClient, RpcError, RpcRequest, RpcRequestHandler};
+use serde_json::{self, Number, Value};
+use solana_sdk::hash::Hash;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::signature::{Keypair, KeypairUtil};
+use solana_sdk::system_transaction::SystemTransaction;
+use solana_sdk::transaction::Transaction;
+use std::error;
+use std::io::{Error, ErrorKind};
+use std::net::SocketAddr;
+
+pub const PUBKEY: &str = "7RoSF9fUmdphVCpabEoefH81WwrW7orsWonXWqTXkKV8";
+pub const SIGNATURE: &str =
+    "43yNSFC6fYTuPgTNFFhF4axw7AfWxB2BPdurme8yrsWEYwm8299xh8n6TAHjGymiSub1XtyxTNyd9GBfY2hxoBw8";
+
+pub struct MockRpu {}
+
+impl RpcRequestHandler for MockRpu {
+    fn make_rpc_request(
+        client: &RpcClient,
+        _id: u64,
+        request: RpcRequest,
+        params: Option<Value>,
+    ) -> Result<Value, Box<dyn error::Error>> {
+        if &client.addr == "fails" {
+            return Ok(Value::Null);
+        }
+        if request == RpcRequest::ConfirmTransaction {
```

Comment:
This code is super noisy.  It looks like it can be written as:

```rust
let val = match request {
    ...
};
Ok(val)
```

And the only place you'd see `return` is on that `RpcRequestError`.  By the way, that `?` with no trailing semicolon is awkward. I'm a little surprised nothing warned you about how the if-else statement ignores that return value.